contract/src/lib.rs

```diff
@@ -327,29 +379,33 @@ impl MpcContract {
         }
     }
 
-    #[allow(unused_variables)]
-    /// `key_version` must be less than or equal to the value at `latest_key_version`
-    pub fn sign(&mut self, payload: [u8; 32], path: String, key_version: u32) -> Promise {
-        let latest_key_version: u32 = self.latest_key_version();
-        assert!(
-            key_version <= latest_key_version,
-            "This version of the signer contract doesn't support versions greater than {}",
-            latest_key_version,
-        );
-        log!(
-            "sign: signer={}, payload={:?}, path={:?}, key_version={}",
-            env::signer_account_id(),
-            payload,
-            path,
-            key_version
-        );
-        match self.pending_requests.get(&payload) {
-            None => {
-                self.add_request(&payload, &None);
-                log!(&serde_json::to_string(&near_sdk::env::random_seed_array()).unwrap());
-                Self::ext(env::current_account_id()).sign_helper(payload, 0)
+    fn get_signature_per_payload(&self, payload: [u8; 32]) -> Option<Option<(String, String)>> {
```

Comment:
nit(rust): we should not have functions with `get_` prefix unless it's `get()` or `get_mut()` functions. So we should have `signature_per_payload` instead